frontend/src/pages/modelCatalog/components/ModelCatalogCards.tsx

```diff
@@ -1,16 +1,31 @@
 import React from 'react';
-import { Gallery, GalleryItem } from '@patternfly/react-core';
+import { Gallery, GalleryItem, Title, Stack, StackItem } from '@patternfly/react-core';
 import { CatalogModel, ModelCatalogSource } from '~/concepts/modelCatalog/types';
 import { ModelCatalogCard } from './ModelCatalogCard';
 
+const ModelCatalogSection: React.FC<{ source: ModelCatalogSource }> = ({ source }) => (
+  <Stack hasGutter>
+    <StackItem>
+      <Title headingLevel="h2">{source.displayName || `${source.source} models`}</Title>
+    </StackItem>
+    <StackItem>
+      <Gallery hasGutter>
+        {source.models.map((model: CatalogModel, index) => (
+          <GalleryItem key={`${source.source}-${index}`}>
+            <ModelCatalogCard model={model} source={source.source} />
+          </GalleryItem>
+        ))}
+      </Gallery>
+    </StackItem>
+  </Stack>
+);
+
 export const ModelCatalogCards: React.FC<{ sources: ModelCatalogSource[] }> = ({ sources }) => (
-  <Gallery hasGutter data-testid="model-catalog-cards">
-    {sources.map((mcs, srcIndex) =>
-      mcs.models.map((cm: CatalogModel, modelIndex) => (
-        <GalleryItem key={`${srcIndex}-${modelIndex}`}>
-          <ModelCatalogCard model={cm} source={mcs.source} />
-        </GalleryItem>
-      )),
-    )}
-  </Gallery>
+  <Stack hasGutter data-testid="model-catalog-cards">
+    {sources.map((source) => (
+      <StackItem key={source.source}>
+        <ModelCatalogSection source={source} />
+      </StackItem>
+    ))}
```

Comment:
We only want to break things into sections if there is more than one source --- at least that's what we wrote up, but actually now that we're not doing the drill-down behavior anymore I'm not sure it matters. It just means we'll have the "Red Hat models" section header even if those are the only models. Maybe that's fine? @yih-wang do you care?

Reply:
> We only want to break things into sections if there is more than one source
+1... But I don’t think having just one section header would cause significant confusion, so I’m fine with it.